lib/recog/nizer.rb

```diff
@@ -41,6 +41,24 @@ def self.match(match_key, match_string)
     nil
   end
 
+  def self.multi_match(match_key, match_string)
```

Comment:
I implemented the second method as multi-match because using `match` with single match returns a single object that is the matched fingerprint.  If we change it to multi match it becomes an array of matched fingerprints.  If we were to change it to always return an Array of fingerprints, I agree we could make it work... but then we break Metasploit.